vassal-app/src/main/java/VASSAL/counters/BasicPiece.java

```diff
@@ -170,11 +186,26 @@ public void setMap(Map map) {
     }
   }
 
+  /** @return Each GamePiece belongs to a single {@link Map} */
   @Override
   public Map getMap() {
     return getParent() == null ? map : getParent().getMap();
   }
 
+  /**
+   * Properties can be associated with a piece -- many may be game-specific, but others
+   * are standard, such as the LocationName property exposed by BasicPiece -- and can
+   * be read through this interface. The properties may or may not need to be encoded in
+   * the piece's {@link #getState} method. Properties include the value of e.g. {@link Marker}
+   * Traits, {@link DynamicProperty} Traits, and so forth. Furthermore they include the values
+   * of any visible "Global Property" in a Vassal module, whether at the module level, map
+   * level, or zone level.
+   *
+   * <br><br>When using this interface a piece's own properties are preferred to those of
+   * "Global Properties", and those in turn are searched Zone-first then Map, then Module.
+   * @param key String key of property to be returned
+   * @return Object containing new value of the specified property
+   */
```

Comment:
Interestingly (and perhaps dangerously), Scratchpad properties currently rank ahead of Persistent properties. I will write out the hierarchy is it is currently implemented since this is a "docs PR", and we can talk about if really Persistent should be checked over Scratchpad (which was also my intuition when I was coming through here before - I was like hunh, that's scary)

Reply:
Actually, that's probably good. It prevents an errant modder from creating a persistent property that puts a scratch-pad property in a shadow.

Reply:
lol. Sounds good. We've obviously gotten this far without the world ending :)